client/src/hooks/helpers/useResources.tsx

```diff
@@ -12,28 +12,29 @@ import {
   type Entity,
 } from "@dojoengine/recs";
 import { useCallback, useEffect, useMemo, useState } from "react";
-import { ProductionManager } from "../../dojo/modelManager/ProductionManager";
+import { ResourceManager } from "../../dojo/modelManager/ResourceManager";
 import { getEntityIdFromKeys } from "../../ui/utils/utils";
 import { useDojo } from "../context/DojoContext";
 import useUIStore from "../store/useUIStore";
 
 export function getResourcesUtils() {
   const {
     setup: {
-      components: { Resource, ResourceCost, Realm, OwnedResourcesTracker },
+      components: { ResourceCost, Realm, OwnedResourcesTracker },
     },
   } = useDojo();
 
   const getResourcesFromBalance = (entityId: ID): Resource[] => {
-    // todo: switch back to items_count when working
+    const currentDefaultTick = useUIStore.getState().currentDefaultTick;
+
     const ownedResources = getComponentValue(OwnedResourcesTracker, getEntityIdFromKeys([BigInt(entityId)]));
     if (!ownedResources) return [];
-    // const resourceIds = getResourceIdsFromPackedNumber(ownedResources.resource_types);
     const resourceIds = resources.map((r) => r.id);
     return resourceIds
       .map((id) => {
-        const resource = getComponentValue(Resource, getEntityIdFromKeys([BigInt(entityId), BigInt(id)]));
-        return { resourceId: id, amount: Number(resource?.balance) || 0 };
+        const resourceManager = new ResourceManager(useDojo().setup, entityId, id);
```

Comment:
We should remove getResourceCosts() as it isn't used anymore and has been replaced by a configManager method.
Also, is there a specific need to call useDojo here again instead of passing it from the first call at the top of function ?